internal/pkg/client/cache/dir.go

```diff
@@ -130,6 +130,15 @@ func NewHandle(cfg Config) (*Handle, error) {
 	if baseDir == "" {
 		baseDir = getCacheBasedir()
 	}
+
+	// create basedir if cache is enabled and it does not exist
+	if _, err := os.Stat(baseDir); os.IsNotExist(err) {
```

Comment:
Would it make sense to move the call to `initCacheDir()` here then, instead of duplicating code? `initCacheDir(baseDir)` should work, right?

Reply:
Yep! :smile: